### ISSUE TITLE
pwndbg fails to work with 64 bit qemu costum kernel devel new issue

### ISSUE THREAD

**Issue by Xyhlon** [pwndbg/pwndbg#1009]
### Description

Basically the same as here #911 

### Steps to reproduce
`git-clone git@github.com:IAIK/sweb.git`
`mkdir -p /tmp/sweb; cd /tmp/sweb; cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 $PROJECT_PATH; make -j; make qemu`
cancel and run
`make debug; make -j; make qemudbg` to activate the debug info
https://www.iaik.tugraz.at/teaching/materials/os/tutorials/prerequisites-installation/
in other terminal:
`make rungdb`
start stepping through the code you will the following messages

### My setup

```pwndbg> show version
GNU gdb (GDB) 11.2
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
```
git commit d348c749

```Traceback (most recent call last):
  File "/usr/share/pwndbg/pwndbg/commands/__init__.py", line 131, in __call__
    return self.function(*args, **kwargs)
  File "/usr/share/pwndbg/pwndbg/commands/__init__.py", line 225, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/usr/share/pwndbg/pwndbg/commands/context.py", line 266, in context
    result[target].extend(func(target=out,
  File "/usr/share/pwndbg/pwndbg/commands/context.py", line 347, in context_regs
    regs = get_regs()
  File "/usr/share/pwndbg/pwndbg/commands/context.py", line 402, in get_regs
    desc = pwndbg.chain.format(value)
  File "/usr/share/pwndbg/pwndbg/chain.py", line 100, in format
    rest.append(M.get(link, symbol))
  File "/usr/share/pwndbg/pwndbg/color/memory.py", line 45, in get
    page = pwndbg.vmmap.find(int(address))
  File "/usr/share/pwndbg/pwndbg/memoize.py", line 52, in __call__
    value = self.func(*args, **kwargs)
  File "/usr/share/pwndbg/pwndbg/vmmap.py", line 78, in find
    for page in get():
  File "/usr/share/pwndbg/pwndbg/memoize.py", line 52, in __call__
    value = self.func(*args, **kwargs)
  File "/usr/share/pwndbg/pwndbg/vmmap.py", line 45, in get
    pages.extend(monitor_info_mem())
  File "/usr/share/pwndbg/pwndbg/memoize.py", line 52, in __call__
    value = self.func(*args, **kwargs)
  File "/usr/share/pwndbg/pwndbg/vmmap.py", line 264, in monitor_info_mem
    assert end-start == size, "monitor info mem output didn't pass a sanity check"
AssertionError: monitor info mem output didn't pass a sanity check```

**Comment by disconnect3d** [comment]
Hey, @Xyhlon can u show output of `monitor info mem` command?

**Comment by disconnect3d** [comment]
@Xyhlon ping

**Comment by Xyhlon** [comment]
sry for the delayed response @disconnect3d .
```
pwndbg> monitor info mem
0000000008000000-0000000008006000 0000000000006000 urw
0000000008009000-000000000800b000 0000000000002000 urw
00007ffffffff000-ffff800000000000 0000000000001000 urw
fffff00000000000-fffff00040000000 0000000040000000 -rw
ffffffff800b8000-ffffffff8017d000 00000000000c5000 -rw
ffffffff80388000-ffffffff8069a000 0000000000312000 -rw
```
this is the output

**Comment by disconnect3d** [comment]
Can you also run pwndbg with `set exception-debugger on` and then do: `print(end, start, size)`?

**Comment by disconnect3d** [comment]
omg, for `00007ffffffff000-ffff800000000000 0000000000001000 urw` we are getting:
```
In [13]: 0xffff800000000000 - 0x00007ffffffff000
Out[13]: 18446462598732845056
```

Certainly not matching the `0x1000` size...

```
In [15]: hex(0x00007ffffffff000 + 0x1000)
Out[15]: '0x800000000000'
```

**Comment by Xyhlon** [comment]
```(Pdb) print(end, start, size)
18446603336221196288 140737488351232 4096
(Pdb) hex(print(end, start, size))
18446603336221196288 140737488351232 4096
*** TypeError: 'NoneType' object cannot be interpreted as an integer
(Pdb) print(hex(end), hex(start),hex(size))
0xffff800000000000 0x7ffffffff000 0x1000
```

Sry dude it seems I forgot what the is and isn't mentioned in the website I linked and it would be probably necessary to run `make debug; make -j; make qemudbg`  before you run `make rungdb` in order to activate all the debug info.

**Comment by disconnect3d** [comment]
TL;DR: It seems we should remove the assertion...

**Comment by Xyhlon** [comment]
ok nice

**Comment by disconnect3d** [comment]
@Xyhlon Hey, any chance you can see if this still reproduces with the latest pwndbg version? We are now using gdb-pt-dump in order to get QEMU kernel memory map information.

**Comment by gsingh93** [comment]
Closing this, if this is still an issue feel free to reopen.
